repos/system_upgrade/common/actors/setsystemdservicesstate/libraries/setsystemdservicesstate.py

```diff
@@ -0,0 +1,26 @@
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import SetSystemdServicesStateTask
+
+
+def _try_set_unit_state(command, unit):
+    try:
+        # it is possible to call this on multiple units at once,
+        # but failing to enable one service would cause others to not enable as well
+        run(['systemctl', command, unit])
+    except CalledProcessError as err:
+        api.current_logger().error('Failed to {} unit "{}". Message: {}'.format(command, unit, str(err)))
+        # TODO(mmatuska) produce post-upgrade report
+
+
+def process():
+    services_to_enable = []
+    services_to_disable = []
```

Comment:
what about to use sets? so we do not try to enable/disable some services multiple times. Also we could check then easily intersections of both sets. We should check whether some tasks results into the requirement to enable & disable some same services in the same time which in such a case we know we cannot fulfill. Regarding the we should raise another error log (later post report). Currently in such a case let's keep the current logic unchanged. So in case of conflict the service will be disabled